### ISSUE TITLE
[Feature Request] Rework `Follow` profile to be a setting instead of a profile so it can be used with others profiles

### ISSUE THREAD

**Issue by ThaDaVos** [openhab/openhab-core#3825]
Currently one can only have the `follow` profile *or* any other profile - you cannot combine them without using Rules - which is some cases, in my opinion, lead to unneeded use of rules.

So my idea is, that the `follow` profile would be replaced with a simple checkbox to allow a channel to follow the states, just as the `follow` profile does now - this way, one can combine the follow functionality (which in my opinion shouldn't be a profile at all) to be used with other more useful profiles like the `Script` transformations or `map`.

Why? Because in some cases, like in mine now, you have multiple channels linked to a single item, these other channels use the `script` transformation (javascript) to modify the value before it's send to the thing (because it's needed) - but now they won't follow the other channel which gets updates from a thing.

My case:
I have an old Openhab 3 and a new Openhab 4 instance - the idea is to rebuild my Openhab and make everything privacy focused, so no Google/Alexa Integration - because I am not the only one living at home, I don't want to break their current voice control until I have a replacement up and running - so I used the `Remote Openhab` binding to link both instances (binding is on OH4 side).

Now the issue, I have a few switches in my model which are used to add ON/OFF functionality to my Tuya based lights - these, when using the Tuya Binding from [here](https://github.com/smarthomej/addons/tree/3.2.x/bundles/org.smarthomej.binding.tuya), does not have a dedicated channel for power - only brightness - so I created two transformations scripts, one from switch to number (brightness) and one for the inverse - so this button/switch (which works from the UI) is used to turn on/off multiple lights or a single light by setting the brightness - now the issue: I hooked up the old switch (which was available when using mqtt-tuya project) in my OH3 to this switch using the channel on the remote-server thing - I did this for other lights and devices and works perfectly but only when you add the `follow` profile to the channel to communicate back the state update from the remote channel - the thing is, I cannot do this for these special channels which use the script profile as a channel can only have a single profile - which got me thinking - why is this a profile and not a option/setting which can be set on the channel instead?

So, tldr: change the `follow` profile to be an option/setting on the channel instead so it can be combined with the other, in my opinion, more useful profiles instead of forcing people to go use rules just to get this simple functionality/scenario to work.

## Your Environment

* Version used: 4-snapshot (as there are issues with the model not updating when adding equipment)
* Environment name and version (e.g. Chrome 111, Java 17, Node.js 18.15, ...): Docker, Firefox
* Operating System and version (desktop or mobile, Windows 11, Raspbian Bullseye, ...): Ubuntu Server, with Docker image (tag openhab/openhab:snapshot)

**Comment by rkoshak** [comment]
> Now the issue, I have a few switches in my model which are used to add ON/OFF functionality to my Tuya based lights - these, when using the Tuya Binding from [here](https://github.com/smarthomej/addons/tree/3.2.x/bundles/org.smarthomej.binding.tuya), does not have a dedicated channel for power - only brightness - so I created two transformations scripts, one from switch to number (brightness) and one for the inverse - so this button/switch (which works from the UI) is used to turn on/off multiple lights or a single light by setting the brightness -

This is unnecessary. You can send ON/OFF commands to a Dimmer Channel. That's why a dedicated Channel doesn't exist for power, it's not needed. Just treat the Dimmer Item as if it were a Switch (use a Switch element on sitemaps, use a toggle widget on MainUI, sendCommand(ON) in rules, etc.) and it will work as a Switch. Same goes for the Color Item which supports both ON/OFF and PercentType commands where it will work like a Switch or a Dimmer respectively. You don't need a separate Switch Item for this and, if I'm reading correctly, therefore don't need the `follow` profile either.

> why is this a profile and not a option/setting which can be set on the channel instead?

Largely because it's the least disruptive way to implement it. To implement it as a setting on the Link(?) would require exposing a bunch of stuff to the Link that is currently unavailable to the Link. 

Note, the only reason Profiles exist at all is because of the follow profile. That was what drove the creation of profiles in the first place. 

Unfortunately, the ability to chain profiles is also not an option. This was explored awhile back to see if we could chain transform profiles similar to how transforms work in the MQTT, HTTP, etc. bindings.

**Comment by ThaDaVos** [comment]
Just a note about the switch item, it was used to turn multiple lights on/off - in this case, all 6 of my tuya ceiling spotlights - as linking/grouping items is not really a thing, as far as I know, especially in regard to the Google integration, I used a switch with multiple channel links instead
